src/models/hooks/activityReport.js

```diff
@@ -73,6 +73,30 @@ const moveDraftGoalsToNotStartedOnSubmission = async (sequelize, instance, optio
   }
 };
 
+const setSubmittedDate = async (sequelize, instance) => {
```

Comment:
If you move the hook from afterUpdate to beforeValidate, then the function could be change from calling another update to just adding the value change to the current update. This will be significantly after as it reduces the number of round trips to the database. For an example of this please look at src/models/hooks/goal.js for lines 29-37 & 147-156. you will need lines 148-150 to be in your beforeValidate. the call to beforeValidate will then also need to be added to the hooks defined in the model.

Reply:
Should be good to go.